crates/primitives/src/transaction/mod.rs

```diff
@@ -1058,6 +1058,22 @@ impl reth_codecs::Compact for TransactionSignedNoHash {
     }
 }
 
+#[cfg(any(test, feature = "reth-codec"))]
+impl reth_codecs::Compact for TransactionSigned {
```

Comment:
imo this makes sense

especially if we do

https://github.com/paradigmxyz/reth/issues/12471

this also isn't more expensive because this just does the clone somewhere else